ntp-proto/src/time_types.rs

```diff
@@ -381,6 +382,47 @@ ntp_duration_scalar_div!(u16);
 ntp_duration_scalar_div!(u32);
 // u64 and usize deliberately excluded as they can result in overflows
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PollInterval(i8);
+
+impl PollInterval {
+    pub const MIN: Self = Self(4);
+    pub const MAX: Self = Self(17);
```

Comment:
The situation here is interesting. The spec says (section 7.2):

    | MINPOLL   | 4     | minimum poll exponent (16 s)     |
    | MAXPOLL   | 17    | maximum poll exponent (36 h)     |

The code skeleton says

    #define MINPOLL         6       /* % minimum poll interval (64 s)*/
    #define MAXPOLL         17      /* % maximum poll interval (36.4 h) */

finally, the ntp repository 

    #define	NTP_MINPOLL	3	/* log2 min poll interval (8 s) */
    #define NTP_MINDPOLL	6	/* log2 default min poll (64 s) */
    #define NTP_MAXDPOLL	10	/* log2 default max poll (~17 m) */
    #define	NTP_MAXPOLL	17	/* log2 max poll interval (~36 h) */

So nobody agrees. Guess we follow the spec literally here?

Reply:
Indeed, decided to go with the main (normative) part of the spec here. We can always decide to deviate here should we want to.